Configurealarms beeprepeat fix and added back missing debugDescription values

Lytrix opened the pull request:
@bbassett who is working on the AndroidAPS implementation found a bug which is fixed with this PR:
In the ConfigureAlerts command, the beep type and beep repeat are listed one way in the document and reversed in the swift code.
Fixed the above and also added these:

- Created enum for BeepRepeat to know what setting you select
- Fixed Tests to use enum values of BeepRepeat and fixed the switched values
- Fixed failing PodStateTest

Lytrix committed:
- Fixed switched BeepRepeat <->BeepType encoding/decoding fields
- Created enum for BeepRepeat to read more clearly what setting you will get
- Fixed Tests to pass the enum values of BeepRepeat

Lytrix committed:
Fixed failing podStateTests: added expiresAt value

ps2 commented on OmniKitTests/PodStateTests.swift:
```diff
@@ -13,7 +13,7 @@ class PodStateTests: XCTestCase {
     func testNonceValues() {
         
-        var podState = PodState(address: 0x1f000000, activatedAt: Date(), timeZone: .currentFixed, piVersion: "1.1.0", pmVersion: "1.1.0", lot: 42560, tid: 661771)
+        var podState = PodState(address: 0x1f000000, activatedAt: Date(), expiresAt: Date().addingTimeInterval(podSoftExpirationTime), timeZone: .currentFixed, piVersion: "1.1.0", pmVersion: "1.1.0", lot: 42560, tid: 661771)
```
It doesn't really matter in this context, but might be good to fix nonetheless. `expiresAt` should be calculated using the same date instance that `activatedAt` uses, otherwise they are different times.

ps2 replied:
and there are two more below

Lytrix committed:
Use the same Date instance on activatedAt and expiresAt parameters

Lytrix committed:
Removed double entries in PodInfoFaultEvent description caused by rebasing

Lytrix committed:
Added missing description values back to debugDescription

Lytrix committed:
Replaced DeliveryType with DeliveryStatus to decode values like 0x06

Lytrix committed:
Updated to deliveryStatus types to pass tests

ps2 commented on OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift:
```diff
@@ -10,10 +10,10 @@ import Foundation
     public var podInfoType: PodInfoResponseSubType = .faultEvents
     public let podProgressStatus: PodProgressStatus
-    public let deliveryType: DeliveryType
+    public let deliveryStatus: StatusResponse.DeliveryStatus
```
I thought DeliveryType was the shared type, so I moved that into Pod.swift.  If DeliveryStatus is the shared type, it should probably move to Pod.swift, and DeliveryType back into the CancelCommand.  Also, DeliveryType doesn't need a description function (the existing one doesn't work for the various combinations of bit fields anyways).

Lytrix replied:
Yes, I discovered it, when using the test command to ask for a 02 fault event. Then it returned values more consistent with the DeliveryStatus. I switched the two types accordingly and removed the description.

ps2 commented on OmniKit/MessageTransport/MessageBlocks/ConfigureAlertsCommand.swift:
```diff
@@ -45,6 +45,18 @@ public struct ConfigureAlertsCommand : NonceResyncableMessageBlock {
         case time(TimeInterval)
     }
+
+    public enum BeepRepeat: UInt8 {
```
Awesome! Love these types. :)

Lytrix replied:
Thanks! Never could have written those so specifically without the hard work of Joe!

Lytrix committed:
- Moved DeliveryStatus to Pod.swift because of it being a shared type.
- Moved DeliveryType to cancelCommand because it is only applied there.
- Removed DeliveryType  description function because the options can be combined.
- Rerouted DeliveryType in to cancelDeliveryCommand in PodCommsSession
